_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,298 @@
+pilot should watch for creation of its xDS cache. This is done by starting the pilot application
+with the additional command line arguments,  `--appNamespace, ns-1`.  Where *ns-1* is the
+namespace that the tenant’s application will be deployed in. An example snippet from the
+istio-system1.yaml file is included below.
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-pilot
+  namespace: istio-system1
+  annotations:
+    sidecar.istio.io/inject: "false"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        istio: pilot
+    spec:
+      serviceAccountName: istio-pilot-service-account
+      containers:
+      - name: discovery
+        image: docker.io/<user ID>/pilot:<tag>
+        imagePullPolicy: IfNotPresent
+        args: ["discovery", "-v", "2", "--admission-service", "istio-pilot", "--appNamespace", "ns-1"]
+        ports:
+        - containerPort: 8080
+        - containerPort: 443
+
+```
+
+#### Deploying the tenant application in a namespace
+Now that the cluster administrator has created the tenant's namespace (ex. *istio-system1*) and
+the pilot's service discovery has been configured to watch for a specific application
+namespace (ex. *ns-1*), create the application manifests to deploy in that tenant's specific
+namespace. Example:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns-1
+```
+And add the namespace reference to each resource type included in the applications manifest
+file.  For example:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: details
+  labels:
+    app: details
+  namespace: ns-1
+```
+Although not shown, the application namespaces will also have RBAC settings limiting access
+to certain resources. These RBAC settings could be set by the cluster administator and/or
+the tenant administator.
+
+#### Using Istioctl commands in a multitenant environment
+When defining route rules (`RouteRule`) or destination policies (`DestinationPolicy`), it is
+necessary to ensure that the `istioctl` command is scoped to
+the `namespace` the Istio control plane is running in to ensure the resource is created
+in the proper namespace. Additionally, the rule itself must be scoped to the tenant's namespace
+so that it will be applied properly to that tenant's mesh.  The "-i" option is used to create
+(or get or describe) the rule in the namespace that the istio control plane is deployed in.
+The "-n" option will scope the rule to the tenant's mesh and should be set to the namespace that
+the tenant's app is deployed in. Note that the -n option can be skipped on the command line if
+the .yaml file for the resource scopes it properly instead.
+
+For example, the following command would be required to add a route rule to the *istio-system1*
+namespace:
+```sh
+istioctl –i istio-system1 create -n ns-1 -f route_rule_v2.yaml
+```
+
+```sh
+istioctl -i istio-system1 -n ns-1 get routerule
```

Comment:
Please put output into a separate block. This allows the user to easily copy the command-line to the clipboard without drgging in all the output.